pkg/pillar/cmd/zedrouter/appnumonunet.go

```diff
@@ -163,14 +166,42 @@ func appNumOnUNetAllocate(ctx *zedrouterContext, baseID uuid.UUID,
 		// for static we pick the topmost numbers so avoid consuming
 		// dynamic IP address from a smallish DHCP range.
 		if isStatic {
-			for i := types.BitMapMax; i >= 0; i-- {
-				if !baseMap.IsSet(i) {
+			niStatus, err := ctx.pubNetworkInstanceStatus.Get(baseID.String())
+			if err != nil {
+				log.Fatalf("appNumOnUNetAllocate: no network status found: %s", err)
+			}
+			networkInstanceStatus := niStatus.(types.NetworkInstanceStatus)
+			// firstly check if provided addr is inside BitMap
+			// XXX we are limited with BitMapMax(255) and cannot cover larger ranges of IP addresses
+			// XXX we must use appNum to check intersections outside of BitMapMax
+			if networkInstanceStatus.DhcpRange.Start != nil &&
+				!networkInstanceStatus.DhcpRange.Start.IsUnspecified() {
+				for i := types.BitMapMax; i >= 0; i-- {
+					if !addr.Equal(types.AddToIP(networkInstanceStatus.DhcpRange.Start, i)) {
+						// calculated IP not match static
+						continue
```

Comment:
Why do we care about a match here? We need some free appnum for the static IPs but no need for them to match AFAICT.

Reply:
We want to not intersect with other IPs (later-created dynamic ones), so we should set particular appNum.

Reply:
But the assumption is that the static IPs are in a different part of the IP subnet than the dynamic ones. I have no idea what will break if you want to allow static IPs inside the DHCP range. So I think we need to fix this at the source by having a sane DHCP range.

We have code in zedagent/parseIpspec which makes sure the DHCP range is sane.
Thus for the default network instance we get
  "Subnet": {
    "IP": "10.1.0.0",
    "Mask": "//8AAA=="
  },
  "Gateway": "10.1.0.1",
  "DhcpRange": {
    "Start": "10.1.0.128",
    "End": "10.1.255.254"
  },

If the issue is that some zedcloud code assumes it can use the last IP address(es) in the subnet, then we should either fix that zedcloud code to use e.g., .2 OR have parseipspec reserve some space at the end of the subnet. 
For the default network instance that would be trivial since we don't need more than 126 IP addresses in the range and right now we have 64000 in that range.